src/Http/Chain.php

```diff
@@ -164,10 +163,8 @@ public function assemble(array $params = [], array $options = [])
 
         $this->assembledParams = [];
 
-        $routes = ArrayUtils::iteratorToArray($this->routes);
-
-        end($routes);
-        $lastRouteKey = key($routes);
+        $routes       = ArrayUtils::iteratorToArray($this->routes);
+        $lastRouteKey = array_key_last($routes);
```

Comment:
Oh nice!

Reply:
yes 👍 , there is rule for that `ArrayKeyFirstLastRector` https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#arraykeyfirstlastrector